include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t              maxVertexInputAttributes;
+    //uint32_t              maxVertexInputBindings;
+    //uint32_t              maxVertexInputAttributeOffset;
+    //uint32_t              maxVertexInputBindingStride;
+    //uint32_t              maxVertexOutputComponents;
+    //uint32_t              maxTessellationGenerationLevel;
+    //uint32_t              maxTessellationPatchSize;
+    //uint32_t              maxTessellationControlPerVertexInputComponents;
+    //uint32_t              maxTessellationControlPerVertexOutputComponents;
+    //uint32_t              maxTessellationControlPerPatchOutputComponents;
+    //uint32_t              maxTessellationControlTotalOutputComponents;
+    //uint32_t              maxTessellationEvaluationInputComponents;
+    //uint32_t              maxTessellationEvaluationOutputComponents;
+    //uint32_t              maxGeometryShaderInvocations;
+    //uint32_t              maxGeometryInputComponents;
+    //uint32_t              maxGeometryOutputComponents;
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
+    //uint32_t              maxFragmentDualSrcAttachments;
+    //uint32_t              maxFragmentCombinedOutputResources;
+    uint32_t maxComputeSharedMemorySize;
+    //uint32_t              maxComputeWorkGroupCount[3];
+    //uint32_t              maxComputeWorkGroupInvocations;
+    uint32_t maxWorkgroupSize[3];
+    //uint32_t              subPixelPrecisionBits;
+    //uint32_t              subTexelPrecisionBits;
+    //uint32_t              mipmapPrecisionBits;
+    //uint32_t              maxDrawIndexedIndexValue;
+    uint32_t maxDrawIndirectCount;
+    //float                 maxSamplerLodBias;
+    float    maxSamplerAnisotropyLog2;
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
+    //uint32_t              viewportSubPixelBits;
+    //size_t                minMemoryMapAlignment;
+    uint32_t bufferViewAlignment;
+    uint32_t UBOAlignment;
+    uint32_t SSBOAlignment;
+    //int32_t               minTexelOffset;
+    //uint32_t              maxTexelOffset;
+    //int32_t               minTexelGatherOffset;
+    //uint32_t              maxTexelGatherOffset;
+    //float                 minInterpolationOffset;
+    //float                 maxInterpolationOffset;
+    //uint32_t              subPixelInterpolationOffsetBits;
+    //uint32_t              maxFramebufferWidth;
+    //uint32_t              maxFramebufferHeight;
+    //uint32_t              maxFramebufferLayers;
+    //VkSampleCountFlags    framebufferColorSampleCounts;
+    //VkSampleCountFlags    framebufferDepthSampleCounts;
+    //VkSampleCountFlags    framebufferStencilSampleCounts;
+    //VkSampleCountFlags    framebufferNoAttachmentsSampleCounts;
```

Comment:
do we want to expose this? I think It's better to differentiate them from maxImageDimensions

Reply:
yes, its `GL_MAX_FRAMEBUFFER_*` in GL and GLES

OpenGL kinda assumes all attachments will have identical sample counts, hence there's only one `GL_MAX_FRAMEBUFFER_SAMPLES` .

obviously "all" sample counts `<=` are supported so you need to `(0x1u<<(1+findMSB(val)))-1` to convert to a bitfield of sample counts

P.S. for the "layers" in GLES you either need ver 3.2 or `EXT_geometry_shader` or `OES_geometry_shader`

Reply:
> P.S. for the "layers" in GLES you either need ver 3.2 or EXT_geometry_shader or OES_geometry_shader
please check If I did this correctly in IOpenGLPhysDev

Reply:
> > P.S. for the "layers" in GLES you either need ver 3.2 or EXT_geometry_shader or OES_geometry_shader
> 
> please check If I did this correctly in IOpenGLPhysDev

all good.